sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -0,0 +1,221 @@
+    {extra_query_args}
+""",
+).strip()
+
+dm_exec_requests_exclude_keys = {
+    'sql_handle',
+    'plan_handle',
+    'statement_sql_handle',
+    'task_address',
+    'page_resource',
+    'scheduler_id',
+    'context_info',
+    # remove status in favor of session_status
+    'status',
+    # remove session_id in favor of id
+    'session_id',
+}
+
+
+def _hash_to_hex(hash):
+    return to_native_string(binascii.hexlify(hash))
+
+
+class SqlserverActivity(DBMAsyncJob):
+    """Collects query metrics and plans"""
+
+    def __init__(self, check):
+        self.check = check
+        self.log = check.log
+        collection_interval = float(check.activity_config.get('collection_interval', DEFAULT_COLLECTION_INTERVAL))
+        if collection_interval <= 0:
+            collection_interval = DEFAULT_COLLECTION_INTERVAL
+        self.collection_interval = collection_interval
+        super(SqlserverActivity, self).__init__(
+            check,
+            run_sync=is_affirmative(check.activity_config.get('run_sync', False)),
+            enabled=is_affirmative(check.activity_config.get('enabled', True)),
+            expected_db_exceptions=(),
+            min_collection_interval=check.min_collection_interval,
+            config_host=check.resolved_hostname,
+            dbms="sqlserver",
+            rate_limit=1 / float(collection_interval),
+            job_name="query-activity",
+            shutdown_callback=self._close_db_conn,
+        )
+        self._conn_key_prefix = "dbm-activity-"
+        self._activity_payload_max_bytes = MAX_PAYLOAD_BYTES
+
+    def _close_db_conn(self):
+        pass
+
+    def run_job(self):
+        self.collect_activity()
+
+    def _get_active_connections(self, cursor):
+        self.log.debug("collecting sql server current connections")
+        self.log.debug("Running query [%s]", CONNECTIONS_QUERY)
+        cursor.execute(CONNECTIONS_QUERY)
+        columns = [i[0] for i in cursor.description]
+        # construct row dicts manually as there's no DictCursor for pyodbc
+        rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+        self.log.debug("loaded sql server current connections len(rows)=%s", len(rows))
+        return rows
+
+    def _get_activity(self, cursor):
+        self.log.debug("collecting sql server activity")
+        extra_query_args = self._get_extra_activity_query_args()
+        query = ACTIVITY_QUERY.format(extra_query_args=extra_query_args)
+        self.log.debug("Running query [%s]", query)
+        cursor.execute(query)
+        columns = [i[0] for i in cursor.description]
+        # construct row dicts manually as there's no DictCursor for pyodbc
+        rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+        return rows
+
+    def _get_extra_activity_query_args(self):
+        extra_query_args = (
+            " AND NOT (sess.status = 'sleeping' AND DATEDIFF(second, at.transaction_begin_time, GETDATE()) < {})"
```

Comment:
Why is this filter here? 

```
NOT DATEDIFF(second, at.transaction_begin_time, GETDATE()) < collection_interval
```

If I'm reading it correctly, does this mean that we're keeping only those transactions that started greater than `collection_interval` seconds ago?

Reply:
no the full filter is 
```
NOT (sess.status = 'sleeping' AND DATEDIFF(second, at.transaction_begin_time, GETDATE()) < {})
```

which means I want everything from the sessions the table that is not sleeping both newer than the collection interval. This means it would catch an open idle transaction, which older than the 10 second collection interval. I just tested this by creating a new transaction with the `sa` user, and then did not call commit 

you can see it shows up here https://dd.datad0g.com/s/yB5yjZ/u2r-atv-aud
![Screen Shot 2021-11-23 at 3 07 05 PM](https://user-images.githubusercontent.com/14317240/143142931-08b4b398-3f1d-4291-8c02-4ebe45f01eac.png)

Reply:
want me to add a comment explaining the purpose of this?